Source/WebKit/WebProcess/WebPage/WebPage.messages.in

```diff
@@ -824,4 +824,6 @@ GenerateSyntheticEditingCommand(enum:uint8_t WebKit::SyntheticEditingCommandType
     CloseCurrentTypingCommand()
 
     SimulateClickOverFirstMatchingTextInViewportWithUserInteraction(String targetText) -> (bool success)
+
+    SetPageScaleFactor(float scaleFactor, WebCore::IntPoint point) -> ()
```

Comment:
We don’t need to introduce a new test-only IPC message for this. You could replace `Messages::WebPageProxy::PageScaleFactorDidChange` in `WebPage::scalePage` with a completion handler and use the existing IPC message (`Messages::WebPage::ScalePage`).

Reply:
As discussed we can't do this as:
1. it's not a 1-1 replacement for the setPageScaleFactor in Internals.(scalePage is doing more).
2. All callers of scalePage don't really have a completionHandler and there are a bunch of them and rooting them out is probably good for another PR. Scope TBD